### PR TITLE
Use xmlmode for svg

### PR DESCRIPTION
**Pull request by bezoerb**

### CODE REVIEW

File: lib/Parser.js [hunk 1]
```diff
@@ -119,6 +119,20 @@ function Parser(cbs, options){
 
 require("util").inherits(Parser, require("events").EventEmitter);
 
+Parser.prototype._checkXmlMode = function(){
```
Comment (fb55) [inline lib/Parser.js]: This should be cached (in `onopentagname`).

File: lib/Parser.js [hunk 2]
```diff
@@ -87,7 +87,32 @@ var voidElements = {
...
+	feOffset: true,
+	feSpecularLighting: true,
+	feTile: true,
+	feTurbulence: true,
+
+	// light source element
+	// http://www.w3.org/TR/SVG/intro.html#TermLightSourceElement
+	feDistantLight: true,
+	fePointLight: true,
+	feSpotLight: true
```
Comment (bezoerb) [inline lib/Parser.js]: I'll need to add a testcase for the additional self-closing svg elements

File: lib/Parser.js [hunk 3]
```diff
@@ -144,6 +170,11 @@ Parser.prototype.onopentagname = function(name){
 		name = name.toLowerCase();
 	}
 
+	if(name.toLowerCase() === "svg" && typeof this._cachedXmlMode === "undefined"){
```
Comment (bezoerb) [inline lib/Parser.js]: better?

**Comment by bezoerb** [comment]
@fb55 did you have a chance yet to take look at the changes?

**Comment by bezoerb** [comment]
@fb55 ping ;)

**Comment by fb55** [comment]
My first impression is that this seems to be always active, when `xmlMode` should definitely disable the behavior (subsequent `svg` tags could be detected via the stack). Besides, I'm not sure this is a worthwhile addition, as my original goal was to move the parser into an entirely different direction. I currently don't have the time to work on this and I'm sorry to leave it hanging.

**Comment by bezoerb** [comment]
@fb55 thanks for the reply. As for now this would definitely fix some problems with inline SVG where cheerio produces an invalid output.

**Comment by bezoerb** [comment]
@fb55 Are there any updates on this or could the svg issue be solved in another way by now?

**Comment by bezoerb** [comment]
@fb55 it seems like this issue is still in place. Would you mind taking another look at the PR?

**Comment by kisenka** [comment]
@fb55 could you please take a look at this PR?

**Comment by fb55** [comment]
My problem with this from 2015 still was not addressed: This is always active, independent of the other settings. This should be disabled by default, with an option to enable it.

**Comment by bezoerb** [comment]
@fb55: I think it should be always active for SVG because there's no need to parse inline SVG with `{xmlMode: false}` as this will result in broken inline SVGs most of the time. (And because of the `_xmlModeStack` the xmlMode will always be set to it's configured value as soon as the parser leaves the SVG.)

Nevertheless i've addressed your concern and made it optional (disabled by default). It will only kick in if the xmlMode is set to `"auto"`

**Comment by topaxi** [comment]
Just had some issues where some of my inline SVGs got scrambled. Any way to get this or another fix merged? I'd be happy to help out.

**Comment by phawxby** [comment]
@fb55 what needs doing to get this merged in? I've just spent the last hour scratching my head to figure out why Cheerio was scrambling SVG's. I agree with @bezoerb in that SVG's should be treated as XML, it is an XML based format.

**Comment by fb55** [comment]
Closing in favor of #254

**Pull request closed without merging**